kore/src/Kore/Strategies/Goal.hs

```diff
@@ -739,8 +740,13 @@ removeDestination goal = errorBracket $ do
         result = Conditional.andPredicate configuration removal
     pure $ makeRuleFromPatterns goal result destination
   where
-    destination = getDestination goal
     configuration = getConfiguration goal
+    configFreeVars = Pattern.freeVariables configuration
+
+    RulePattern { right, ensures } = toRulePattern goal
+    right' = topExistsToImplicitForall configFreeVars right
+    destination =
+        Pattern.withCondition right' (Conditional.fromPredicate ensures)
```

Comment:
I think I already asked you, so sorry for asking again, but why do we keep the ensures here? Also this applies to the function `getDestination` as well (it's at the bottom of this file and we use it in several places in this module and in the `Verification` module as well I think).